### PR TITLE
APP-273 implement script to create initial daos

### PR DESCRIPTION
**Pull request by Rekard0**
## Description

Please include a summary of the change and be sure you follow the contributions rules we do provide [here](./CONTRIBUTIONS.md)

Task: [ID](https://aragonassociation.atlassian.net/browse/APP-273)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

File: .github/workflows/contracts-deploy.yml [hunk 1]
```diff
@@ -72,24 +72,39 @@ jobs:
     needs: ['build-deploy']
     steps:
       - uses: actions/checkout@v2
-        with:
```
Comment (mathewmeconry) [inline .github/workflows/contracts-deploy.yml:75]: don't delete this

File: .github/workflows/contracts-deploy.yml [hunk 2]
```diff
@@ -72,24 +72,39 @@ jobs:
...
+        run: node .github/helpers/contracts/dummy-dao/proposal.ts rinkeby ${{ secrets.ETH_KEY }} erc20
+      - name: Proposal of Whitelist
+        run: node .github/helpers/contracts/dummy-dao/proposal.ts rinkeby ${{ secrets.ETH_KEY }}
       - name: Commit changes
         run: |
           git fetch
           git pull
-          git config --global user.name "Arabot-1"
-          git config --global user.email "arabot-1@users.noreply.github.com"
+          git config --global user.name "Rekard0"
```
Comment (mathewmeconry) [inline .github/workflows/contracts-deploy.yml:104]: revert it back to arabot

File: .github/workflows/contracts-deploy.yml [hunk 3]
```diff
@@ -72,24 +72,39 @@ jobs:
...
+        run: yarn install --pure-lockfile
+      - name: Compile
+        run: cd packages/contracts && yarn run build
       - name: Update Releases.md
         run: node .github/helpers/contracts/updateReleasesMD.js
       - name: Update active_contracts.json
         run: node .github/helpers/contracts/updateActiveContractsJSON.js
       - name: Update Subgraph manifests
         run: node .github/helpers/subgraph/updateManifest.js
+      - name: create Dummy DAOs erc20
```
Comment (mathewmeconry) [inline .github/workflows/contracts-deploy.yml:88]: move these scripts after the commit. we wanna make sure to have the updated data in the repo and than try to create these daos

File: .github/workflows/contracts-deploy.yml [hunk 4]
```diff
@@ -72,24 +72,39 @@ jobs:
...
+        run: node .github/helpers/contracts/dummy-dao/createDao.ts rinkeby ${{ secrets.ETH_KEY }} erc20
```
Comment (mathewmeconry) [inline .github/workflows/contracts-deploy.yml:89]: node cannot run .ts files. use ts-node for it

**Review by mathewmeconry — CHANGES_REQUESTED** [review]

File: .github/workflows/contracts-deploy.yml [hunk 5]
```diff
@@ -85,27 +87,35 @@ jobs:
...
+          git config --global user.name "Arabot-1"
+          git config --global user.email "arabot-1@users.noreply.github.com"
+          git add active_contracts.json
+          git add packages/contracts/Releases.md
+          git add ./packages/subgraph/	
+          git commit -am "Updates files with deployed contract addresses"
+          git push
       - name: create Dummy DAOs erc20
-        run: node .github/helpers/contracts/dummy-dao/createDao.ts rinkeby ${{ secrets.ETH_KEY }} erc20
+        run: ts-node .github/helpers/contracts/dummy-dao/createDao.ts $NETWORK ${{ secrets.ETH_KEY }} erc20
```
Comment (mathewmeconry) [inline .github/workflows/contracts-deploy.yml:102]: from where does it get the $NETWORK from? 
if you like to do it per network you need to move these scripts into the `build-deploy` step or setup a matrix job here.
I recommend the matrix job. See `build-deploy` as a reference

**Review by mathewmeconry — CHANGES_REQUESTED** [review]

File: packages/contracts/deploy/verify.ts [hunk 6]
```diff
@@ -74,7 +74,7 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
   await verifyContract(distributorBase, []);
 };
 export default func;
-func.tags = ['DAOFactory', 'TokenFactory', 'Registry', 'verify'];
+func.tags = ['Verify'];
```
Comment (mathewmeconry) [inline packages/contracts/deploy/verify.ts:77]: this changes breaks the verify functionalitiy of `contracts-deploy.yml` workflow

File: .github/helpers/contracts/dummy-dao/createDao.ts [hunk 7]
```diff
@@ -44,30 +45,64 @@ async function createDao() {
...
-  const metadata = ethers.utils.hexlify(ethers.utils.toUtf8Bytes(cid.path));
-  const daoConfig = [daoName, metadata];
-  const votingSettings = ['200000000000000000', '400000000000000000', '60000'];
+  let metadata = ethers.utils.hexlify(ethers.utils.toUtf8Bytes(cid.path));
+  let daoConfig = [daoName, metadata];
+  let votingSettings = ['200000000000000000', '400000000000000000', '60000'];
+
+  let overrides = {};
+
+  if (networkName === 'mumbai') {
```
Comment (mathewmeconry) [inline .github/helpers/contracts/dummy-dao/createDao.ts:54]: move this into its own file and replace all occurrences with an import and function call to this new file.
Also, add this gas estimation for polygon mainnet too and integrate it into the `hardhat.config.ts` so we can remove the hardcoded gas fee in `networks.json`

**Review by mathewmeconry — CHANGES_REQUESTED** [review]

File: packages/contracts/deploy/verify.ts (hunk 6 above)
Comment (Rekard0) [reply packages/contracts/deploy/verify.ts:77]: contracts-deploy.yml is also updated , it should not break it, did u looked at https://github.com/aragon/core/blob/f/APP-273-start-testnet-with-dummy-dao/.github/workflows/contracts-deploy.yml#L58

Comment (mathewmeconry) [reply packages/contracts/deploy/verify.ts:77]: :+1:  have you checked that it follows the right order of tasks?
I had issues before with it

Comment (Rekard0) [reply packages/contracts/deploy/verify.ts:77]: rearranged the order and it seems to deploy fine

File: .github/helpers/contracts/dummy-dao/createDao.ts (hunk 7 above)
Comment (Rekard0) [reply .github/helpers/contracts/dummy-dao/createDao.ts]: - moved it out.
- i couldn't find a way to async update hardhat config with gas data, 
do u think it is possible?

**Pull request merged**